src/components/Button/Button.tsx

```diff
@@ -1,29 +1,58 @@
-import React from 'react';
+import React, { MouseEventHandler } from 'react';
+import classNames from 'classnames';
+
+import Spinner from '../utils/Spinner/Spinner';
 
 import styles from './Button.module.css';
 
 export interface Props {
   children: string;
-  type: 'primary' | 'secondary';
-  onClick?: (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
+  type?: 'primary' | 'secondary';
+  onClick?: MouseEventHandler<HTMLButtonElement>;
+  className?: string;
+  isDisabled?: boolean;
+  isLoading?: boolean;
+  prefixIcon?: React.ReactElement;
+  suffixIcon?: React.ReactElement;
   name?: string;
 }
 
-const Button: React.FC<Props> = ({ children, type = 'primary', onClick, name }: Props) => {
-  const getAppearanceClass = (buttonType: typeof type) => {
-    if (buttonType === 'secondary') {
-      return styles.typeSecondary;
-    }
-    return styles.typePrimary;
-  };
-
-  const classNames: string[] = [styles.base];
+const Button: React.FC<Props> = ({
+  children,
+  type = 'primary',
+  onClick,
+  className,
+  isDisabled,
+  isLoading,
+  prefixIcon,
+  suffixIcon,
+  name,
+}: Props) => {
+  const buttonClassNames = classNames(
+    styles.base,
+    {
+      [styles.typePrimary]: (!isDisabled || !isLoading) && type === 'primary',
```

Comment:
We don't have to do this logic here. We can keep it simple here and let the css do the logic of what styles should have priority. If `.disabled` is placed last in the css file then the `.primary` styles be will be overwritten.